File: src/main/java/com/snackgame/server/applegame/domain/game/Percentile.java
package com.snackgame.server.applegame.domain.game;

import com.snackgame.server.applegame.exception.InaccuratePercentileException;

public class Percentile {

    private final double percentile;

    public Percentile(double percentile) {
        validateRangeOf(percentile);
        this.percentile = percentile;
    }

    private void validateRangeOf(double percentile) {
        if (percentile < 0 || 1 < percentile) {
            throw new InaccuratePercentileException(percentile);
        }
    }

    public int percentage() {
        double percentageWise = Math.floor(percentile * 100);
        return (int)percentageWise;
    }
}

File: src/main/java/com/snackgame/server/game/snackgame/core/domain/Percentile.kt
package com.snackgame.server.game.snackgame.core.domain

import com.snackgame.server.game.snackgame.exception.InaccuratePercentileException
import kotlin.math.floor

class Percentile(private val percentile: Double) {

    init {
        validateRangeOf(percentile)
    }

    private fun validateRangeOf(percentile: Double) {
        if (percentile < 0 || percentile > 1) {
            throw InaccuratePercentileException(percentile)
        }
    }

    fun percentage(): Int {
        val percentageWise = floor(percentile * 100)
        return percentageWise.toInt()
    }
}
